doc/source/building/macosx.rst

```diff
@@ -0,0 +1,188 @@
+===============================
+Building From Source on Mac OSX
+===============================
+
+These instructions describe how to build NumPy and SciPy libraries from
+source.
+
+If you just want to use NumPy or SciPy, install pre-built binaries as described
+in :doc:`/install`.
+
+Python
+------
+
+Apple ships its own version of Python with OS X. However, we
+*strongly* recommend installing the `official Python distribution
+<https://www.python.org/downloads/>`__.
+
+Alternatively, use Python from one of the OS X package managers 
+(Homebrew, MacPorts, Fink).  
+
+Apple's Developer Tools
+-----------------------
+
+Apple's Developer Tools provide a number of key libraries,
+particularly the `vecLib Framework
+<https://developer.apple.com/library/mac/documentation/Performance/Conceptual/vecLib/>`__
+, which includes the BLAS and LAPACK libraries for optimizing matrix and vector
+operations. The most recent version should be included on your OS X installation CD.
+Ensure that all components are installed by choosing **customize** when available
+during the install process and selecting all optional packages - at least the X11
+development tools and (on OS X 10.6 or lower) the 10.4 SDK.
+
+Compilers (C/C++/FORTRAN/Cython)
+--------------------------------
+
+Though virtually any commercial C/C++ compiler may be used with SciPy, OS X
+comes with GNU C compilers pre-installed. The only thing missing is the GNU
+FORTRAN compiler.
+
+We recommend gfortran; this is a free, open source, F95 compiler. We suggest you
+use the following binaries:
+
+* For Snow Leopard : https://cran.r-project.org/bin/macosx/tools/gfortran-4.2.3.pkg
+* For Lion : http://r.research.att.com/gfortran-lion-5666-3.pkg (for
+  Xcode 4.1)
+* Later versions : http://r.research.att.com/tools/gcc-42-5666.3-darwin11.pkg (for Xcode
+  4.2 or higher) (also available through Homebrew)
```

Comment:
I think the better recommendation is Homebrew now, has gfortran 4.9 that works well.